Add a note about APP_PATHS not being populated by default (#18265)
dotnet/docs: docs/core/dependency-loading/default-probing.md

Finally the environment variable `ADDITIONAL_DEPS` can be used to add additional dependencies.

> Properties `APP_PATHS` and `APP_NI_PATHS` are not populated by default and will be left out for most applications.

Reviewer: ```suggestion
The `APP_PATHS` and `APP_NI_PATHS` properties are not populated by default and are omitted for most applications.
```

Slight rewording suggestion for localizability.
